auth/grpctransport/grpctransport.go

```diff
@@ -240,37 +239,37 @@ func dial(ctx context.Context, secure bool, opts *Options) (*grpc.ClientConn, er
 	return grpc.DialContext(ctx, endpoint, grpcOpts...)
 }
 
-// grpcTokenProvider satisfies https://pkg.go.dev/google.golang.org/grpc/credentials#PerRPCCredentials.
-type grpcTokenProvider struct {
-	auth.TokenProvider
+// grpcCredentialProvider satisfies https://pkg.go.dev/google.golang.org/grpc/credentials#PerRPCCredentials.
```

Comment:
I would rename to `grpcCredentialsProvider`, since `Credentials` is the type name.